src/lib/rectangleTool.ts

```diff
@@ -92,3 +100,64 @@ export function updateRectangleSketch(
     createLiteral(Math.abs(y)), // This will be the height of the rectangle
   ])
 }
+
+/**
+ * Mutates the pipeExpression to update the center rectangle sketch
+ * @param pipeExpression
+ * @param x
+ * @param y
+ * @param tag
+ */
+export function updateCenterRectangleSketch(
+  pipeExpression: PipeExpression,
+  deltaX: number,
+  deltaY: number,
+  tag: string,
+  originX: number,
+  originY: number
+) {
+  let startX = originX - Math.abs(deltaX)
+  let startY = originY - Math.abs(deltaY)
+
+  // pipeExpression.body[1] is startProfileAt
+  ;((pipeExpression.body[1] as CallExpression)
+    .arguments[0] as ArrayExpression) = createArrayExpression([
```

Comment:
Should this make use of your new cool type narrowing functions as well?

Reply:
Yup, let me do this. I forgot since this was not complaining about typescript errors.

Reply:
Addressed.